src/reanimated2/NativeMethods.ts

```diff
@@ -17,11 +17,18 @@ export function measure(
   animatedRef: RefObjectFunction<Component>
 ): MeasuredDimensions | null {
   'worklet';
-  if (!_WORKLET || !isNative) {
+  if (!isNative) {
     console.warn(
       '[Reanimated] measure() cannot be used for web or Chrome Debugger'
     );
     return null;
+  } else if (!_WORKLET) {
+    console.warn(
+      '[Reanimated] measure() was called from the wrong thread. This may happen ' +
+        'if you called measure() in the useAnimatedStyle hook. If you want to' +
+        'prevent this warning then wrap the call with `if (_WORKLET)`.'
```

Comment:
1. Instead of "wrong thread" use something like "main JS context".
2. Explain that `measure` is only available in the UI runtime and cannot be used from main JS context.
3. Technically, it's the worklet inside `useAnimatedStyle`, not the hook itself.
4. Explain that `useAnimatedStyle` first calls given worklet from main JS runtime during render.
5. Explain why someone needs to add `if (_WORKLET)` (so that `measure` is called only from the UI context).